app/models/project_role.rb

```
@@ -0,0 +1,6 @@
+class ProjectRole < ActiveRecord::Base
+  belongs_to :project, required: true
+  belongs_to :role, required: true
+
+  validates :project_id, uniqueness: { scope: :role_id }
```

I agree the error message is not the best, but this is really an edge case anyway. For the user who's adding a role to a project, it should probably just fail silently in the UI. I'm not sure we should spend a ton of time on it if it's not a really obvious problem. I'm guessing it's a 1/1000+ thing.